### PR TITLE
Chore: document conventional commits

### PR DESCRIPTION
**Pull request by drewbo**
## Changes proposed in this pull request:
- Adds documentation around our use of [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) close https://github.com/cloud-gov/product/issues/2230

## security considerations
None

### CODE REVIEW

**Comment by drewbo** [comment]
Note that the only "truly new" portion of this PR is the [bottom section of the new DEVELOPMENT.md document](https://github.com/cloud-gov/pages-core/pull/4188/files#diff-fc2a310fcfedfefb0046def697f932def80cbb1e78c689bc0af3c8eab3e33eceR452)

**Review by apburnes — DISMISSED** [review]
📝

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -0,0 +1,473 @@
...
+        Status: "Enabled",
+      }
+    ]
+  }
+}
+```
+
+## Commits, Releases, and Deployment
+
+All commit messages on the default branch should follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) specification. In addition, an associated ticket number should be added where relevant. Examples:
```
Comment (svenaas) [inline DEVELOPMENT.md:454]: It's not that unusual for one of our commits to need to reference a ticket on one of our other GitHub repos. Should this perhaps clarify what those should look like?

Comment (svenaas) [reply DEVELOPMENT.md:454]: For example, the runbook I just did was a ticket on this repo, but the implementation was a commit to a different repo.

File: DEVELOPMENT.md [hunk 2]
```diff
@@ -0,0 +1,473 @@
...
+- [This ruleset](https://github.com/cloud-gov/pages-core/settings/rules/26912) checks the commit message against a regex rule before it can be merged to the default branch.
+- No [merge commit pull requests](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/about-merge-methods-on-github) are allowed to the repo
+- The default branch enforces a [linear commit history](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-rulesets/available-rules-for-rulesets#require-linear-history)
+
+The benefit of adhering to this convention is that we can more easily reason about our commit history and also generate nice changelogs via [Release Please](https://github.com/googleapis/release-please).
+
+If you didn't follow this convention while making your commits locally or on a development, you'll still have (at least) two opportunities to edit the commit history to match the Convention Commits specification:
+- while the code is on a non-default branch, you can perform an [interative rebase](https://git-scm.com/docs/git-rebase) to rewrite the history.
```
Comment (svenaas) [inline DEVELOPMENT.md:470]:
```suggestion
- while the code is on a non-default branch, you can perform an [interactive rebase](https://git-scm.com/docs/git-rebase) to rewrite the history.
```

File: DEVELOPMENT.md [hunk 3]
```diff
@@ -0,0 +1,473 @@
...
+- prior to merging a PR, you can select either "Squash and merge" or "Rebase and merge" which will change the commit message(s) in the PR.
```
Comment (svenaas) [inline DEVELOPMENT.md:471]: It **is** confusing that both of these are characterized as merges whereas merge commits on the feature branch are not allowed. This isn't particularly friendly, but I'm not sure how to make it better.

**Review by svenaas — CHANGES_REQUESTED** [review]
One copyedit and a few suggestions.

**Review by apburnes — APPROVED** [review]

**Review by svenaas — APPROVED** [review]

**Pull request merged**